### ISSUE TITLE
tf2center.com

### ISSUE THREAD

**Issue by SpennyTF** [jspenguin2017/AdBlockProtector#141]
Would be great if you find a way to block it.

https://tf2center.com/lobbies

**Comment by X01X012013** [comment]
Hi, can you describe the problem? I don't see anything wrong with that website.

**Comment by uBlock-user** [comment]
@X01X012013 Better add a format for reporting issues here.

**Comment by uBlock-user** [comment]
I cannot repoduce but if you want to add then this will be the patch -  `a.readOnly("adblockAllowed", true);`

**Comment by SpennyTF** [comment]
The test link is: http://tf2center.com/lobbies
Opening the site with adblocker enabled redirects you to https://tf2center.com/adblock
Adding `a.readOnly("adblockAllowed", true);` didn't work for me

**Comment by uBlock-user** [comment]
It's based on cookies. Remove all your cookies and site data and it will work.

**Comment by X01X012013** [comment]
I am not redirected, it works fine here. Can you post your setup (browser, script host, and ad blocker)?

**Comment by SpennyTF** [comment]
I have tried clearing my cookie and using the bait, but it still redirected me. As it seems they don't redirect non logged in users. I'm using uBlock Origin, Tampermonkey and Chrome

**Comment by jspenguin2017** [comment]
Hum... OK, I'll have another look.

**Comment by jspenguin2017** [comment]
Is a different file served when you are logged in? Can you dump the page code?

**Comment by SpennyTF** [comment]
There is once 
`const noAds=false;`
and what you have already found: https://tf2center.com/assets/javascript/advert.js

**Comment by jspenguin2017** [comment]
@SpennyTF 

> and what you have already found: https://tf2center.com/assets/javascript/advert.js

That file is not blocked. 

> `const noAds=false;`

Eh... Unfortunately, we wouldn't be able to change this one. For technical reasons (let me know if you want to know). 

Can you dump page source? I think the page served is different if you are logged in.

**Comment by uBlock-user** [comment]
It's read-only right ? Cannot be reassigned like we do with `var`

**Comment by jspenguin2017** [comment]
No, if we assign before them, it causes syntax error in their code and crashes the whole block. If we try to assign it after, our code will crash. The new syntax is kind of hard to deal with, it doesn't work in every browser so we kind of got away with it for two years. Now browsers start to catch up and we'll see ECMA 2015 standards in more webs pages... 
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/const

**Comment by jspenguin2017** [comment]
`const` will not create a property to `window`, so I'm not sure if a setter/getter would do it.

**Comment by SpennyTF** [comment]
I have put it into a gist for you
https://gist.github.com/SpennyTF/1ae9c6c7a9a148432c8fc9fedaf5f86b

**Comment by jspenguin2017** [comment]
Thanks. Found the problem. A fix is added to Script. 

Offending code: 
```JavaScript
    setInterval(function() {
        if (C()) {
            window.location = "/adblock"
        }
    }, 3000);
```
Solution: 
```JavaScript
if (a.domCmp(["tf2center.com"])) {
    //Issue: https://github.com/jspenguin2017/AdBlockProtector/issues/141
    a.filter("setInterval", /\"\/adblock\"/);
}
```

**Comment by SpennyTF** [comment]
thanks!

**Comment by jspenguin2017** [comment]
Did you update Script to 6.168? If so, please do this: 

1. Enter settings page, turn on debug mode
2. Dump logs from the console
3. Turn off debug mode

**Comment by jspenguin2017** [comment]
Hum... I don't really have a way to test it, so it's kind of guesswork... Let me try again...

**Comment by jspenguin2017** [comment]
OK, I think we fixed it for good this time. Added to Script 6.169. 
```JavaScript
if (a.domCmp(["tf2center.com"])) {
    a.filter("setInterval", /\"\/adblock\"/);
    a.filter("setTimeout", /^function\(\)\{B\(F\+1\)\}$/);
}
```

**Comment by SpennyTF** [comment]
http://i.imgur.com/pjttPMz.png
Its weird, seems like it still executing the function. (It redirected me again)

**Comment by jspenguin2017** [comment]
Can you edit the script and add a space here? 
![image](https://cloud.githubusercontent.com/assets/7283682/25106199/880113b8-2385-11e7-8dbe-db4e3aab9119.png)

**Comment by SpennyTF** [comment]
Yeah that seems to have fixed it!

**Comment by jspenguin2017** [comment]
OK great. 
The change is added to Script.

**Comment by SpennyTF** [comment]
Thank you so much!

**Comment by jspenguin2017** [comment]
@uBlock-user 
> @X01X012013 Better add a format for reporting issues here. 

Sorry I missed your comment. It might be a good idea, but most of the time all we need is a link. In this case, we didn't know that you need to login to trigger the redirect, so having a template would not help. 

@SpennyTF 
> By the way is there a way to donate some cash to you as thanks?

Thank you for considering donating, but I do not accept donation currently. There are many other ways you can help us out, including telling your friend about this project, reporting websites that are detecting ad blockers, and it looks like you have some technical knowledge about web development, there could be issues that you can contribute. Or even just testing solutions out can help, some times Geo Lock and log-in conditions make testing hard, and a diverse community is important for proper testing. 

Looks like this issue is fixed, I'll close it now.

**Comment by uBlock-user** [comment]
@jspenguin2017 Can't we use `a.noAccess` on it ? That way don't allow to be created at all.

**Comment by jspenguin2017** [comment]
@uBlock-user Again, it will crash the whole block which breaks the page.

**Comment by uBlock-user** [comment]
So how do you neutralize `const` ?

**Comment by jspenguin2017** [comment]
@uBlock-user I don't know. We'll find other weaknesses. Keep in mind that `const` can't be edited, it can't be edited by the web page's script neither. So chances are it's fine. 
The real problem is the keyword `let`. It doesn't create a property to `window` and would cause syntax error if the identifier is declared before. Unlike `const`, the identifier declared with `let` can have its value changed. 
Without tampering with V8, the JavaScript engine, I don't think there are much we can do with Userscript.
